draft-ietf-httpbis-safe-method-w-body.xml

```diff
@@ -285,27 +285,35 @@ q=foo&amp;limit=10&amp;sort=-published
 
   <section title="The &quot;Accept-Query&quot; Header Field" anchor="field.accept-query">
     <t>
-      The "Accept-Query" response header field &MAY; be used by a resource to
+      The "Accept-Query" response header field can be used by a resource to
       directly signal support for the QUERY method while identifying
       the specific query format media type(s) that may be used.
     </t>
-<sourcecode type="abnf">
-Accept-Query = 1#media-type
-</sourcecode>
     <t>
-      The Accept-Query header field specifies a comma-separated listing of media
-      types (with optional parameters) as defined by
-      <xref target="HTTP" section="8.3.1"/>. <cref>field syntax currently discussed in <eref target="https://github.com/httpwg/http-extensions/issues/2860"/></cref>
+      "Accept-Query" contains a list of media types (<xref target="HTTP" section="8.3.1"/>),
```

Comment:
sorry, i missed this on first review. here you're saying "a list of media types" with reference to the "Media Types" section of RFC 9110. however, "Media Types" doesn't allow wildcards like `image/*`. for that you need "Media Ranges".

Reply:
That's intentionally.

Reply:
a justification for the intentional "explicit media types, no wildcards" (either in the Issue or here) would be appreciated.

Reply:
We discussed this in the other ticket: what's the use case? Can you give an example?

Reply:
sorry, i'm not finding where a decision to not allow wildcards was made. i saw you mention `*/*` in https://github.com/httpwg/http-extensions/issues/2860#issuecomment-2335436391 , and in https://github.com/httpwg/http-extensions/issues/2860#issuecomment-2452378251 you list as an open question
> how does a server say that it supports any format? Does this make any sense at all? We currently require one value, and IMHO we can stick with that

which i took to mean "at least one value".

a use case that comes immediately to mind is a fancy AI™-powered image search that can accept any image type (at least as "any" as a web browser that tells a server it accepts any image type):
```
Accept-Query: image/*
```

a possible use case for `*/*` could be a query-by-value of a collection/container resource: given any kind of serialization of any kind of object, answer my contained URI of a resource (and maybe metadata about it) if i happen to contain a resource matching that serialization and media type, or `204` if i don't.

Reply:
I'm not convinced by these use cases, for "image/*" I would expect that more than the image would be needed, thus some kind of multipart would be used. For "*/*", the situation seems similar.

With that said, I'd propose the simplest possible way to get this resolved would be to indeed allow these values.